tests/platform_tests/test_kdump.py

```diff
@@ -0,0 +1,117 @@
+                                              xcvr_skip_list, reboot_type=REBOOT_TYPE_COLD)
+
+    @pytest.fixture(autouse=True)
+    def tearDown(self, duthosts, enum_rand_one_per_hwsku_hostname,
+                 localhost, pdu_controller):
+        yield
+        # If the SSH connection is not established, or any critical process is exited,
+        # try to recover the DUT by PDU reboot.
+        duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+        dut_ip = duthost.mgmt_ip
+        hostname = duthost.hostname
+        if not self.check_ssh_state(localhost, dut_ip, SSH_STATE_STARTED):
+            if pdu_controller is None:
+                logging.error("No PDU controller for {}, failed to recover DUT!".format(hostname))
+                return
+            self.pdu_reboot(pdu_controller)
+            # Waiting for SSH connection startup
+            pytest_assert(self.check_ssh_state(localhost, dut_ip, SSH_STATE_STARTED, SSH_STARTUP_TIMEOUT),
+                          'Recover {} by PDU reboot failed'.format(hostname))
+            # Wait until all critical processes are healthy.
+            wait_critical_processes(duthost)
+            self.wait_lc_healthy_if_sup(duthost, duthosts, localhost)
+
+    def test_kernel_panic(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost,
+                          conn_graph_facts, xcvr_skip_list):
+        duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+        dut_ip = duthost.mgmt_ip
+        hostname = duthost.hostname
+        dut_datetime = duthost.get_now_time()
+
+        out = duthost.command('show kdump config')
+        if "Enabled" not in out["stdout"]:
+            pytest.skip('DUT {}: Skip test since kdump is not enabled'.format(hostname))
+
+        cmd = 'nohup bash -c "sleep 5 && echo c > /proc/sysrq-trigger" &'
+        res = duthost.shell(cmd)
+        if not res.is_successful:
+            pytest.fail('DUT {} run command {} failed'.format(hostname, cmd))
+
+        # Waiting for SSH connection shutdown
+        pytest_assert(self.check_ssh_state(localhost, dut_ip, SSH_STATE_ABSENT, SSH_SHUTDOWN_TIMEOUT),
+                      'DUT {} did not shutdown'.format(hostname))
+        # Waiting for SSH connection startup
+        pytest_assert(self.check_ssh_state(localhost, dut_ip, SSH_STATE_STARTED, SSH_STARTUP_TIMEOUT),
+                      'DUT {} did not startup'.format(hostname))
+        # Wait until all critical processes are healthy.
+        wait_critical_processes(duthost)
+        self.wait_lc_healthy_if_sup(duthost, duthosts, localhost, conn_graph_facts, xcvr_skip_list)
+        # Verify DUT uptime is later than the time when the test case started running.
+        dut_uptime = duthost.get_up_time()
+        pytest_assert(dut_uptime > dut_datetime, "Device {} did not reboot".format(hostname))
+
+        out = duthost.command('show reboot-cause')
+        if "Kernel Panic" not in out["stdout"]:
```

Comment:
can you use the common proc - check_reboot_cause from common/reboot.py

Reply:
Should I add 'Kernel Panic' to ```reboot_ctrl_dict``` ?

Reply:
Updated.